### PR TITLE
test(ecocredit/core): add project update simulations

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/server/marketplace/utils.go [hunk 1]
```diff
@@ -159,6 +159,17 @@ func (k Keeper) fillOrder(ctx context.Context, sellOrder *api.SellOrder, buyerAc
 			return err
 		}
 		supply.RetiredAmount = supplyRetired.String()
+
+		supplyTradable, err := math.NewDecFromString(supply.TradableAmount)
```
Comment (aleem1314) [inline x/ecocredit/server/marketplace/utils.go:163]: After these two txns
```
{"entry_kind":"msg","height":24,"order":296,"operation":{"route":"/regen.ecocredit.marketplace.v1.MsgSell","name":"/regen.ecocredit.marketplace.v1.MsgSell","comment":"","ok":true,"msg":{"orders":[{"ask_price":{"amount":"24","denom":"stake"},"batch_denom":"C01-41900513-41900613-001","expiration":"4191-05-15T12:45:06Z","quantity":"33"}],"owner":"regen14c3j672kvw9l5uleh4x9uds2fre5vl7y60salu"}}}
{"entry_kind":"msg","height":24,"order":306,"operation":{"route":"/regen.ecocredit.marketplace.v1.MsgBuyDirect","name":"/regen.ecocredit.marketplace.v1.MsgBuyDirect","comment":"","ok":true,"msg":{"buyer":"regen1exnqjlk9g477w80vqvm7scda7l4fcjzfyvx5rf","orders":[{"bid_price":{"amount":"49","denom":"stake"},"quantity":"33","retirement_jurisdiction":"AQ","sell_order_id":"1"}]}}}
```
Invariants are breaking with the following error

```
panic: invariant broken: ecocredit: batch-supply invariant
        tradable supply is incorrect for 1 credit batch, expected 100, got 67

```

Comment (technicallyty) [reply x/ecocredit/server/marketplace/utils.go:163]: I can have a fix for this up today

Comment (ryanchristo) [reply x/ecocredit/server/marketplace/utils.go:163]: Can you open an issue for this @technicallyty for tracking? :pray:

Comment (technicallyty) [reply x/ecocredit/server/marketplace/utils.go:163]: https://github.com/regen-network/regen-ledger/issues/1067

**Review by technicallyty — APPROVED** [review]
lgtm! failing test should be fixed by #1069.

File: x/ecocredit/simulation/operations.go [hunk 2]
```diff
@@ -136,6 +142,18 @@ func WeightedOperations(
 		},
 	)
 
+	appParams.GetOrGenerate(cdc, OpWeightMsgUpdateProjectAdmin, &weightMsgUpdateProjectAdmin, nil,
+		func(_ *rand.Rand) {
+			weightMsgUpdateProjectAdmin = WeightUpdateClass
```
Comment (ryanchristo) [inline x/ecocredit/simulation/operations.go:147]: Should we have a separate weight for update project?

File: x/ecocredit/simulation/operations.go [hunk 3]
```diff
@@ -136,6 +142,18 @@ func WeightedOperations(
...
+		},
+	)
+
+	appParams.GetOrGenerate(cdc, OpWeightMsgUpdateProjectMetadata, &weightMsgUpdateProjectMetadata, nil,
+		func(_ *rand.Rand) {
+			weightMsgUpdateProjectMetadata = WeightUpdateClass
```
Comment (ryanchristo) [inline x/ecocredit/simulation/operations.go:153]: Same here. Should we have a separate weight for update project?

**Review by ryanchristo — APPROVED** [review]
Pre-approving.

**Pull request merged**